okhttp/src/main/java/okhttp3/internal/connection/RealConnectionPool.kt

```diff
@@ -14,249 +14,270 @@
-import okhttp3.Address;
-import okhttp3.Route;
-import okhttp3.internal.Util;
-import okhttp3.internal.connection.Transmitter.TransmitterReference;
-import okhttp3.internal.platform.Platform;
+import okhttp3.Address
+import okhttp3.Route
+import okhttp3.internal.Util
+import okhttp3.internal.Util.closeQuietly
+import okhttp3.internal.connection.Transmitter.TransmitterReference
+import okhttp3.internal.platform.Platform
+import java.io.IOException
+import java.net.Proxy
+import java.util.ArrayDeque
+import java.util.concurrent.SynchronousQueue
+import java.util.concurrent.ThreadPoolExecutor
+import java.util.concurrent.TimeUnit
 
-import static okhttp3.internal.Util.closeQuietly;
+class RealConnectionPool(
+  /** The maximum number of idle connections for each address.  */
+  private val maxIdleConnections: Int,
+  keepAliveDuration: Long,
+  timeUnit: TimeUnit
+) {
+  private val keepAliveDurationNs: Long = timeUnit.toNanos(keepAliveDuration)
 
-public final class RealConnectionPool {
-  /**
-   * Background threads are used to cleanup expired connections. There will be at most a single
-   * thread running per connection pool. The thread pool executor permits the pool itself to be
-   * garbage collected.
-   */
-  private static final Executor executor = new ThreadPoolExecutor(0 /* corePoolSize */,
-      Integer.MAX_VALUE /* maximumPoolSize */, 60L /* keepAliveTime */, TimeUnit.SECONDS,
-      new SynchronousQueue<>(), Util.threadFactory("OkHttp ConnectionPool", true));
-
-  /** The maximum number of idle connections for each address. */
-  private final int maxIdleConnections;
-  private final long keepAliveDurationNs;
-  private final Runnable cleanupRunnable = () -> {
-    while (true) {
-      long waitNanos = cleanup(System.nanoTime());
-      if (waitNanos == -1) return;
-      if (waitNanos > 0) {
-        long waitMillis = waitNanos / 1000000L;
-        waitNanos -= (waitMillis * 1000000L);
-        synchronized (RealConnectionPool.this) {
-          try {
-            RealConnectionPool.this.wait(waitMillis, (int) waitNanos);
-          } catch (InterruptedException ignored) {
-          }
+  private val cleanupRunnable = object : Runnable {
+    override fun run() {
+      while (true) {
+        val waitNanos = cleanup(System.nanoTime())
+        if (waitNanos == -1L) return
+        try {
+          this@RealConnectionPool.waitNanos(waitNanos)
+        } catch (ignored: InterruptedException) {
         }
       }
     }
-  };
-
-  private final Deque<RealConnection> connections = new ArrayDeque<>();
-  final RouteDatabase routeDatabase = new RouteDatabase();
-  boolean cleanupRunning;
+  }
 
-  public RealConnectionPool(int maxIdleConnections, long keepAliveDuration, TimeUnit timeUnit) {
-    this.maxIdleConnections = maxIdleConnections;
-    this.keepAliveDurationNs = timeUnit.toNanos(keepAliveDuration);
+  private val connections = ArrayDeque<RealConnection>()
+  val routeDatabase = RouteDatabase()
```

Comment:
Add TODOs to make these `internal` once all Java call-sites are converted? Or not critical since this file is in `okhttp3/internal`?

Reply:
Yeah, no urgency on that.